x-pack/filebeat/module/suricata/eve/config/eve.yml

```diff
@@ -6,15 +6,180 @@ paths:
+      ignore_failure: true
+      mode: rename
+      fields:
+        - {from: suricata.eve.src_ip, to: source.address}
+        - {from: suricata.eve.src_port, to: source.port, type: long}
+        - {from: suricata.eve.dest_ip, to: destination.address}
+        - {from: suricata.eve.dest_port, to: destination.port}
+        - {from: suricata.eve.proto, to: network.transport}
+  - convert:
+      ignore_missing: true
+      ignore_failure: true
+      mode: copy
       fields:
-        source_ip: json.src_ip
-        source_port: json.src_port
-        destination_ip: json.dest_ip
-        destination_port: json.dest_port
-        transport: json.proto
+        - {from: source.address, to: source.ip, type: ip}
+        - {from: destination.address, to: destination.ip, type: ip}
+        - {from: '@timestamp', to: event.created}
+  - timestamp:
+      field: suricata.eve.timestamp
+      layouts:
+        - '2006-01-02T15:04:05.999999999Z0700' # ISO8601
+  - drop_fields:
+      fields:
+        - suricata.eve.timestamp
+{{ if .community_id }}
+  - community_id:
 {{ end }}
+  - if:
+      equals.suricata.eve.event_type: dns
+    then:
+      - convert:
+          ignore_missing: true
+          ignore_failure: true
+          mode: copy
+          fields:
+            - {from: suricata.eve.dns.id, to: dns.id, type: string}
+            - {from: suricata.eve.dns.rcode, to: dns.response_code}
+            - {from: suricata.eve.dns.type, to: dns.type}
+      - convert:
+          when.equals.dns.type: query
+          ignore_missing: true
+          ignore_failure: true
+          mode: copy
+          fields:
+            - {from: suricata.eve.dns.rrname, to: dns.question.name}
+            - {from: suricata.eve.dns.rrtype, to: dns.question.type}
+      # Handle the version=1 EVE DNS answer format. Each JSON event contains
+      # a single resource record from the DNS response.
+      - script:
+          when.and:
+            - equals.dns.type: answer
+            - not.has_fields: [suricata.eve.dns.version]
+          id: suricata_dns_answers_v1
+          lang: javascript
+          source: >
+            function process(evt) {
+                var name = evt.Get("suricata.eve.dns.rrname");
+                var data = evt.Get("suricata.eve.dns.rdata");
+                var type = evt.Get("suricata.eve.dns.rrtype");
+                var ttl = evt.Get("suricata.eve.dns.ttl");
+
+                var answer = {};
+                if (name) {
+                    answer.name = name;
+                }
+                if (data) {
+                    answer.data = data;
+                }
+                if (type) {
+                    answer.type = type;
+                }
+                if (ttl) {
+                    answer.ttl = ttl;
+                }
+
+                if (Object.keys(answer).length === 0) {
+                    return;
+                }
+                evt.Put("dns.answers", [answer]);
+            }
```

Comment:
The SIEM app has two DNS queries:
1. `Top DNS domains` table: Terms agg on `dns.question.etld_plus_one`, sub-aggs on `dns.question.name`, `source.bytes`, and `destination.bytes`
2. `DNS queries` KPI: Checks if `dns.question.name` exists among other things.

Since there is no filter on just queries or just answers I think we will miscount if there are several documents (for query and answer or one per answer).